src/main/java/org/folio/qm/controller/interceptor/LoggingRequestInterceptor.java

```diff
@@ -0,0 +1,69 @@
+package org.folio.qm.controller.interceptor;
+
+import java.time.Instant;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.integration.XOkapiHeaders;
+
+@Log4j2
+@Component
+@ConditionalOnProperty(
+  prefix = "folio.logging.request",
+  name = "enabled"
+)
+@RequiredArgsConstructor
+public class LoggingRequestInterceptor extends HandlerInterceptorAdapter {
+
+  private static final String START_TIME_ATTR = "startTime";
+  private static final String REQ_ID_ATTR = "okapiReqId";
+
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response,
+                           @NonNull Object handler) {
+    var startTime = Instant.now().toEpochMilli();
+    var requestId = getRequestId();
+    request.setAttribute(START_TIME_ATTR, startTime);
+    request.setAttribute(REQ_ID_ATTR, requestId);
+    log.info("---> [{}] {} {} {}",
+      requestId,
+      request.getMethod(),
+      request.getRequestURI(),
+      request.getQueryString()
+    );
```

Comment:
would be nice to see here and with completion the body of request/response
similarly to feign settings, make sense to enable/disable body output with a configuration setting